src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
+        return NGX_OK;
```

Comment:
@agentzh Thanks for reply~
I intend to let it continue to follow `bodyfilter`. My idea is to exit in the bodyfiler phase, a bit of the scene is likely to be the output of the general body, so I hope that before the output of all the accumulated buffer. [`Ngx_OK` can be used here](https://github.com/openresty/lua-nginx-module/pull/1009/files#diff-a6f1336f42adee1a8437f5073832da13R159) to pass the rest of the bodyfiler phase (such as ngx_http_gzip_body_filter), output all of the remaining content, to prevent clients from accessing any content. [return `ngx.ERROR` here](https://github.com/openresty/lua-nginx-module/pull/1009/files#diff-a6f1336f42adee1a8437f5073832da13R275) prevent again into the bodyfilter stage, behind the content is no longer the output.